android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageView.kt

```diff
@@ -0,0 +1,703 @@
+								)
+							}
+						},
+						value = endButtonText,
+						readOnly = true,
+						onValueChange = { }
+					)
+				}
+
+				Spacer(
+					modifier = Modifier.weight(weight = 2.0f)
+				)
+
+				RoundedTextField(
+					modifier = Modifier
+						.fillMaxWidth()
+						.height(dimensionResource(id = R.dimen.xxxxxxl))
+						.border(
+							dimensionResource(id = R.dimen.xxxs),
+							textFieldBorderColor,
+							RoundedCornerShape(dimensionResource(id = R.dimen.s))
+						),
+					placeholder = {
+						Text(
+							text = stringResource(id = R.string.description),
+							color = secondaryTextColor,
+						)
+					},
+					label = {
+						Text(
+							text = stringResource(id = R.string.description),
+							color = secondaryTextColor,
+						)
+					},
+					shape = RoundedCornerShape(dimensionResource(id = R.dimen.s)),
+					singleLine = false,
+					value = uiState.description,
+					onValueChange = {
+						viewModel.descriptionStream.onNext(it)
+					}
+				)
+
+				Spacer(
+					modifier = Modifier.weight(weight = 2.0f)
+				)
+
+				Text(
+					text = stringResource(id = R.string.upload_images),
+					style = MaterialTheme.typography.titleSmall
+				)
+
+				Spacer(
+					modifier = Modifier.weight(weight = 1.0f)
+				)
+
+				UploadImages(bitmap)
+
+				Spacer(
+					modifier = Modifier.weight(weight = 2.0f)
+				)
+
+				if (openBottomSheet) {
+					DatePickerBottomSheet(bottomSheetState, dateRangePickerState, onDismissRequest = { openBottomSheet = false },
+						onClick = {
+							coroutineScope.launch { bottomSheetState.hide() }.invokeOnCompletion {
+								if (!bottomSheetState.isVisible) {
+									openBottomSheet = false
+								}
+								if (dateRangePickerState.selectedStartDateMillis != null) {
+									startButtonText =
+										displayDateFormatter.formatDate(dateRangePickerState.selectedStartDateMillis, locale = Locale.getDefault())!!
+									val startDate = SimpleDateFormat("MM/dd/yyyy").parse(startButtonText)
+									viewModel.startDateStream.onNext(
+										if (startDate is Date)
+											storeDateFormat.format(startDate)
+										else uiState.startDate)
+								} else {
+									startButtonText = ""
+									viewModel.startDateStream.onNext("")
+								}
+								if (dateRangePickerState.selectedEndDateMillis != null) {
+									endButtonText =
+										displayDateFormatter.formatDate(dateRangePickerState.selectedEndDateMillis, locale = Locale.getDefault())!!
+									val endDate = SimpleDateFormat("MM/dd/yyyy").parse(endButtonText)
+									viewModel.endDateStream.onNext(
+										if (endDate is Date)
+											storeDateFormat.format(endDate)
+										else uiState.endDate)
+								} else {
+									endButtonText = ""
+									viewModel.endDateStream.onNext("")
+								}
+							}
+						})
+				}
+
+				if (bitmap.value != null) {
+					val base64ImageString = bitmap.value!!.toBase64String()
```

Comment:
It's fine for now since demo, but can you add a TODO to do this computation in the `ViewModel` on a `Schedulers.compuation` thread